pkg/controller/core/workload_controller.go

```diff
@@ -398,10 +399,11 @@ func workloadStatus(w *kueue.Workload) string {
 // As a result, the pod's Overhead is not always correct. E.g. if we set a non-existent runtime class name to
 // `pod.Spec.RuntimeClassName` and we also set the `pod.Spec.Overhead`, in real world, the pod creation will be
 // rejected due to the mismatch with RuntimeClass. However, in the future we assume that they are correct.
-func (r *WorkloadReconciler) handlePodOverhead(log logr.Logger, wl *kueue.Workload) {
-	ctx := context.Background()
-
-	for i := range wl.Spec.PodSets {
+func (r *WorkloadReconciler) handlePodOverhead(ctx context.Context, log logr.Logger, wl *kueue.Workload) {
+	for i, ps := range wl.Spec.PodSets {
+		if ps.Template == nil {
```

Comment:
I suppose this is temporary?
We should get the runtime class from the pod template.